src/core/thread/network_data_service.hpp

```diff
@@ -613,6 +613,13 @@ class Manager : public InstanceLocator, private NonCopyable
                        bool        aServerStable,
                        uint8_t &   aServiceId) const;
     Error IterateToNextServer(Iterator &aIterator) const;
+
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+    bool IsBackboneRouterPrimaryPriorTo(const ServerTlv &                 aServerTlv,
```

Comment:
I actually also favor the existing `IsPriorTo` name by @simonlingoogle since we are comparing two items and I feel the current name reads better, i.e., it is clear that we are checking whether the first item "is prior to" the second item. With `IsPreferred()` it is not immediately clear which one is preferred, first or second.

Reply:
While I agree that the ordering of arguments can be ambiguous, it's not clear to me whether something being "prior" means it is preferred or not. How about something like "`IsBackboneRouterPreferredTo()`?

Reply:
`IsBackboneRouterPreferredTo` reads good to me.

Reply:
Renamed to `IsBackboneRouterPreferredTo`.